src/view/MixerElementSelectorWindow.vala

```diff
@@ -0,0 +1,128 @@
+/*
+ * brickman -- Brick Manager for LEGO MINDSTORMS EV3/ev3dev
+ *
+ * Copyright (C) 2016 Kaelin Laundry <wasabifan@outlook.com>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ * MixerElementSelectorWindow.vala - Lists ALSA mixer elements
+ */
+
+using Ev3devKit;
+using Ev3devKit.Ui;
+
+namespace BrickManager {
+
+    public class MixerElementSelectorWindow : BrickManagerWindow {
+        Ui.Menu element_menu;
+
+        public signal void mixer_elem_selected (IMixerElementViewModel selected_element);
+
+        private class MixerElementWithSignalIds: Object {
+            public IMixerElementViewModel element;
+            public ulong notify_signal_handler_id = 0;
+            public ulong button_press_signal_handler_id = 0;
+        }
+
+        public MixerElementSelectorWindow () {
+            title = "Audio mixer elements";
+            element_menu = new Ui.Menu ();
+            content_vbox.add (element_menu);
+        }
+
+        protected string get_element_label_text(IMixerElementViewModel element) {
+            string mute_string = (element.can_mute && element.is_muted) ? ", muted" : "";
+            return "[%u] %s (%ld%%%s)".printf(element.index, element.name, element.volume, mute_string);
```

Comment:
The index isn't just an array index; it is part of the identity of the element. You can have multiple devices with the same index, and you can have multiple devices with the same name. So we need to show the index otherwise it won't always be possible to tell the difference between two devices.